### PR TITLE
feat(macros): add macros for collectors with custom registry

### PR DESCRIPTION
**Pull request by moustafab**
This adds macros for static instantiation of collectors using a custom registry.

Addresses #386

### CODE REVIEW

**Comment by moustafab** [comment]
@lucab @breeswish just thought I'd see if I could bump this to the top of your queues for review. Thanks!

**Comment by moustafab** [comment]
@lucab @breeswish Friendly reminder!

File: src/macros.rs [hunk 1]
```diff
@@ -161,6 +161,43 @@ macro_rules! register_counter {
     }};
 }
 
+/// Create a [`Counter`] and registers to a custom registry.
+///
+/// # Examples
+///
+/// ```
+/// # #[macro_use] extern crate prometheus;
```
Comment (lucab) [inline src/macros.rs:169]: I think you can turn this (and the rest of the examples too) into a `use prometheus::{register_counter_with_registry, opts}` and make it part of the example.

**Comment by lucab** [comment]
@moustafab thanks, it looks great! I've left a minor comment for the doc-example. Can you please rebase this on master?

File: src/macros.rs (hunk 1 above)
Comment (moustafab) [reply src/macros.rs]: Done

**Comment by moustafab** [comment]
@lucab rebased and hopefully ready to merge!

**Pull request merged**